src/type_key.jl

```diff
@@ -0,0 +1,76 @@
+"""
+    type_key(x)
+
+Computes a value to use as a seed for computing the hash value of a type.
+"""
+function type_key end
+
+function type_key(m::Module, h::UInt)
+    if m === Main || m === Base || m === Core
+        return h
+    else
+        mp = parentmodule(m)
+        if mp === m
+            return h
+        else
+            return hash(nameof(m), type_key(mp, h))
+        end
+    end
+end
+
+function type_key(t::UnionAll, h::UInt)
+    h = hash(h, 0xbd5f3e4941dba79d)
```

Comment:
Might want to note where these numerical constants come from. Are they random "nothing up my sleeve"-ish numbers or based on something?

Reply:
Noted.